### PR TITLE
Keep tokenizer workers off the base GPU under CPU feature transport

### LINKED ISSUE #40072: [Bug] Tokenizer workers open CUDA contexts on the base GPU despite --mm-feature-transport cpu
- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

## Describe the bug

`--mm-feature-transport cpu` (the default when unset) is intended to keep tokenizer workers off the base GPU, but three code paths in the tokenizer worker process still each initialize a sticky CUDA context on it:

1. **nvJPEG image decode.** `BaseMultimodalProcessor.gpu_image_decode = True` (`python/sglang/srt/multimodal/processors/base_processor.py:214`) is passed unconditionally to `load_image` in `_load_single_item` (`base_processor.py:970`). On a CUDA host, JPEG bytes are decoded with `decode_jpeg(..., device="cuda")` (`python/sglang/srt/utils/common.py:1852`), which opens a CUDA context in the worker process even though the tensor is later copied back to CPU for transport.

2. **`pin_memory()` on decoded video frames.** `VideoDecoderWrapper.get_frames_as_tensor` and `_parallel_decode` pin CPU output tensors (`python/sglang/srt/utils/video_decoder.py:152,156,192`), and `preprocess_video` pins again in `python/sglang/srt/multimodal/processors/qwen_vl.py:312` and `python/sglang/srt/multimodal/processors/ernie45_vl.py:184,208`. `pin_memory()` on a CPU tensor initializes a CUDA context in the calling process even though the frames travel over the CPU transport.

3. **`frame_count_limit` binds into `use_gpu`.** Both the tokenizer path (`base_processor.py:983`) and the encoder path (`python/sglang/srt/disaggregation/encoder/preprocessor.py:352`) call `load_video(data, frame_count_limit)`, but the second positional parameter of `load_video` is `use_gpu` (`python/sglang/srt/utils/common.py:1981`). The frame limit is never applied (frame sampling is done by the model processors), and the decode device instead depends on an unrelated limit value: a falsy limit silently decodes on CPU, a positive limit would silently enable NVDEC in the tokenizer worker.

Observed impact: a VLM with TP=4 on GPUs 4–7, `--tokenizer-worker-num 6` and the default (cpu) transport — each tokenizer worker accumulated a ~600 MiB sticky CUDA context on the base GPU (GPU 0) as it handled JPEG/video requests, visible in `nvidia-smi` and as `/dev/nvidia-uvm` mappings in the workers' `/proc/<pid>/maps`.

## Reproduction

1. Launch any VLM with tokenizer workers and the default (CPU) transport:

```bash
python3 -m sglang.launch_server \
    --model-path Qwen/Qwen2.5-VL-7B-Instruct \
    --tp 4 \
    --tokenizer-worker-num 4
```

2. Send one JPEG image request and one video request.

3. For each tokenizer worker PID, a CUDA context shows up despite the CPU transport:

```bash
nvidia-smi   # worker PIDs listed on the base GPU with non-zero memory
grep -c nvidia-uvm /proc/<worker-pid>/maps   # non-empty
```

Minimal mechanism repro (any CUDA host, no server needed):

```python
import torch

def nvidia_maps():
    return [l.rsplit()[-1] for l in open("/proc/self/maps") if "nvidia" in l]

print(nvidia_maps())              # []
torch.zeros(10, 10).pin_memory()
print(nvidia_maps())              # ['/dev/nvidia-uvm', ...]
```

## Environment

Code paths verified at `main` commit `826d5170ae` (2026-09-18); the affected production box is not reachable from where this was investigated, so a full `python3 -m sglang.check_env` dump is unavailable. Observed on a multi-GPU node running a 27B-FP8 VLM with TP=4 on GPUs 4–7, `--tokenizer-worker-num 6`, and the default `--mm-feature-transport` (cpu). The minimal repro above only requires torch on any CUDA host.

### PR DESCRIPTION
**Pull request by jodrv**
## Motivation

Fixes #40072. `--mm-feature-transport cpu` (the default) is meant to keep tokenizer workers off the base GPU, but nvJPEG image decode, `pin_memory()` on video frames, and an accidental NVDEC binding each still open a sticky CUDA context in the worker process. This gates all three on the transport that is already selected (no new flag)

## Modifications

- `base_processor._load_single_item`: CPU decode + unpinned frames when the transport is cpu; nvJPEG + pinning kept for `cuda_ipc`/`cuda_vmm`. The new `feature_transport_uses_gpu()` helper falls back to CPU when the mm config namespace is unpublished, so bare unit tests never touch the GPU.
- `load_video` / `VideoDecoderWrapper`: explicit `pin` flag (default `True`, existing callers unchanged). The tokenizer/encoder paths passed `frame_count_limit` positionally into `use_gpu`; frame sampling already happens in the model processors, so the decode device is now passed explicitly.
- Encoder preprocessor decodes with explicit `use_gpu=True` (it owns its GPU).
- `pin_memory()` sites in `qwen_vl` / `ernie45_vl` gated by the same helper.

Net effect: with CPU transport, no tokenizer worker opens a CUDA context on the base GPU; GPU-transport behavior is unchanged.

## Accuracy Tests

No model-output change: under CPU transport the decoded pixels are the same (the tokenizer video path was already decoding on CPU accidentally); GPU transports are untouched. New CPU unit tests: `test/registered/unit/multimodal/test_feature_transport_decode_gating.py`.

## Speed Tests and Profiling

TODO. 
The observable effect is that tokenizer workers under CPU transport no longer hold a ~600 MiB CUDA context on the base GPU.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #35305507419](https://github.com/sgl-project/sglang/actions/runs/35305507419)
Latest PR Test (Extra): :x: [Run #35305507209](https://github.com/sgl-project/sglang/actions/runs/35305507209)
Latest PR Test (AMD ROCm 10): :x: [Run #35305507229](https://github.com/sgl-project/sglang/actions/runs/35305507229)

### CODE REVIEW

**Comment by jodrv** [comment]
@codex review
